src/Aspire.Hosting/Publishing/ManifestPublishingContext.cs

```diff
@@ -299,10 +299,17 @@ public async Task WriteContainerAsync(ContainerResource container)
         WriteBindings(container);
     }
 
-    private void WriteBuildContext(ContainerResource container)
+    private async Task WriteBuildContextAsync(ContainerResource container)
     {
         if (container.TryGetAnnotationsOfType<DockerfileBuildAnnotation>(out var annotations) && annotations.Single() is { } annotation)
         {
+            // If there's a factory, generate the Dockerfile content and write it to the specified path
+            if (annotation.DockerfileFactory is not null)
+            {
+                var dockerfileContent = await annotation.DockerfileFactory(CancellationToken).ConfigureAwait(false);
+                await File.WriteAllTextAsync(annotation.DockerfilePath, dockerfileContent, CancellationToken).ConfigureAwait(false);
```

Comment:
For generated docker file content, we want to copy it to a resource specific path and make sure the dockerfile property in the manifest is updated. That will make to so that the docker file isn't in a random folder while publishing the manifest.